docs/enhancements/image-handling-service.md

```diff
@@ -0,0 +1,166 @@
+
+## Motivation
+
+Most the assisted service functionality is based on many quick API calls with
+relatively small responses. Handling images is all about larger chunks of data and
+potentially longer operations. Additionally, when configured to use PVs as image
+storage, the service would be difficult to scale [depending on the type of storage
+configured](https://docs.openshift.com/container-platform/4.7/storage/understanding-persistent-storage.html#pv-access-modes_understanding-persistent-storage)
+on the cluster. Splitting these responsibilities would allow us to manage and scale
+these deployments separately.
+
+### Goals
+
+* Create and serve discovery ISO images from a separate service
+* Make minimal changes to existing (or proposed) APIs
+
+### Non-Goals
+
+* Serve a discovery image directly to a BMC
+
+## Proposal
+
+The new service (image service) will handle customizing and serving the customized
+image to users. It will query assisted service to retrieve the information that needs
+to be added to the image based on the information passed in the request by the user.
+
+The image service will expose a single API endpoint to download an image. Initially
+this should be compatible with the existing `GET /clusters/{cluster_id}/downloads/image`
+assisted service endpoint.
+
+The assisted service will retain the `POST /clusters/{cluster_id}/downloads/image`
+endpoint which will save the image information to its database and make it available
+to the image service. In the case of the kube API, the controllers will continue
+to save image information from the various CRs as they do currently.
+
+### User Stories
+
+This change should be transparent to users. As such, stories will focus on
+deployment and management concerns rather than user-facing changes.
+
+#### Story 1
+
+As an application operator, I can easily scale the assisted service application
+regardless of my persistent storage type.
+
+#### Story 2
+
+An additional deployment and service will be managed for all deployment methods.
+This includes SaaS, operator, and CI (minikube). Whether an additional image
+is still an open question.
+
+### Implementation Details/Notes/Constraints
+
+#### Authentication and Authorization
+
+The image service will not need to implement authentication or authorization
+directly as it doesn't manage any user data. The user will provide a token to
+the image service and the image service will directly pass that token to
+assisted service on each call. Assisted service will validate the token before
+giving the image service any user information.
+
+#### Template Images
+
+The image service will take on the responsibility of managing the RHCOS template
+images and creating the minimal ISO template images. This means that the image
+service will require persistent storage (when not using S3). This also means
+that assisted service will no longer require persistent storage and will be able
+to scale more easily in more environments.
```

Comment:
The assisted service will still require persistent storage for configuration files and logs.

Reply:
Right, forgot about that :facepalm: